### PR TITLE
Add histogram-based logging for long-running files

### PR DESCRIPTION
**Pull request by rsdy**
Add behind the `debug` feature flag an option to log severe outliers during analytics, including file size.

This is useful, because it seems like large files are hogging 1 thread, and can take up to 40s to complete (e.g. Kubernetes has a lot of generated `.go` files of several MiB in size). Identifying outliers will be helpful in addressing the issue.

If this seems useful, we may want to convert this into a default feature, as the runtime impact should be fairly lightweight.

### CODE REVIEW

**Review by nerdypepper — APPROVED** [review]
until we can improve scope-queries to operate faster on larger files (which i do think we can), a short term solution would be to prevent calculating intelligence on files that take too long to parse (which probably means they contain tons of tokens; e.g.: generated and/or minified).

File: server/bleep/Cargo.toml [hunk 1]
```diff
@@ -89,6 +87,7 @@ replace_with = "0.1.7"
 relative-path = "1.7.2"
 git2 = "0.15.0"
 dunce = "1.0.3"
+histogram = { version = "0.6.9", optional = true }
```
Comment (ggordonhall) [inline server/bleep/Cargo.toml:90]: Can you add a comment explaining that this is used for logging when the debug feature is enabled?

File: server/bleep/src/indexes/file.rs [hunk 2]
```diff
@@ -330,126 +347,153 @@ impl Indexer<File> {
...
+            .match_indices('\n')
+            .flat_map(|(i, _)| u32::to_le_bytes(i as u32))
+            .collect::<Vec<_>>();
+
+        let lines_avg = buffer.len() as f64 / buffer.lines().count() as f64;
+        let last_commit = repo_info.last_commit_unix_secs;
+
+        trace!("writing document");
+
+        let buf_size = buffer.len();
```
Comment (ggordonhall) [inline server/bleep/src/indexes/file.rs:459]:
```suggestion
        #[cfg(feature = "debug")]
        let buf_size = buffer.len();
```

File: server/bleep/Cargo.toml (hunk 1 above)
Comment (rsdy) [reply server/bleep/Cargo.toml:90]: `optional` dependencies are usually picked up by `features`, like here: https://github.com/BloopAI/bloop/pull/9/files#diff-f8919bb455c366daf0d703faef73f63e745e0a06dde47f6e62b5eb88c78687b1R10

**Pull request merged**